apollo-federation/src/query_graph/extract_subgraphs_from_supergraph.rs

```diff
@@ -2080,15 +2087,15 @@ fn remove_inactive_applications(
     let mut replacement_directives = Vec::new();
     let field = object_or_interface_field_definition_position.get(schema.schema())?;
     for directive in field.directives.get_all(name_in_schema) {
-        let (fields, parent_type_pos) = match directive_kind {
+        let (fields, parent_type_pos, is_requires_field_set) = match directive_kind {
             FieldSetDirectiveKind::Provides => {
                 let fields = federation_spec_definition
                     .provides_directive_arguments(directive)?
                     .fields;
                 let parent_type_pos: CompositeTypeDefinitionPosition = schema
                     .get_type(field.ty.inner_named_type().clone())?
                     .try_into()?;
-                (fields, parent_type_pos)
+                (fields, parent_type_pos, false)
```

Comment:
I would instead write `(fields, parent_type_pos, &schema)` here and `(fields, parent_type_pos, &supergraph_schema)` below. Then you don't need the if/else to populate `fields` later.